From cffcf22082879939bbcfa89e88057beda829a0a7 Mon Sep 17 00:00:00 2001
From: David Faure <david.faure@kdab.com>
Date: Sat, 3 Sep 2016 18:01:40 +0200
Subject: gcc6: set -fno-delete-null-pointer-checks for all files

Not just v8, as v8_no_delete_null_pointer_checks does.
This is necessary e.g. for PaintLayer::enclosingSelfPaintingLayer
which also compares this with null.

I'll remove v8_no_delete_null_pointer_checks in a later commit
to avoid a temporary breakage until qtwebengine is updated.

Change-Id: I8782fadc8fab125b5394bd60857e70c6ae75f9b0
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
(cherry picked from commit c42f53ec3ca90bfd98577988462ed51464feebe3)
---
 chromium/build/common.gypi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/3rdparty/chromium/build/common.gypi b/src/3rdparty/chromium/build/common.gypi
index b6a0284..d768ab3 100644
--- a/src/3rdparty/chromium/build/common.gypi
+++ b/src/3rdparty/chromium/build/common.gypi
@@ -4702,6 +4706,9 @@
               '-Wl,--disable-new-dtags',
             ],
           }],
+          [ 'no_delete_null_pointer_checks==1', {
+              'cflags_cc': [ '-fno-delete-null-pointer-checks' ],
+          }],
           ['clang==0', {
             'target_conditions': [
               ['_toolset=="target"', {
-- 
cgit v1.0-4-g1e03

From ae73e50e7d8c1371ed293c10150646a2719f4598 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@theqtcompany.com>
Date: Wed, 14 Sep 2016 17:40:50 +0200
Subject: FIXUP: gcc6: set -fno-delete-null-pointer-checks for all files

Set a default and pull in the value from configuration, and
remove the old v8 only version.

Change-Id: Ica9f971baa28c717178eb2acd074607424792e1d
Reviewed-by: David Faure <david.faure@kdab.com>
---
 chromium/build/common.gypi       | 4 ++++
 chromium/v8/build/toolchain.gypi | 7 -------
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/src/3rdparty/chromium/build/common.gypi b/src/3rdparty/chromium/build/common.gypi
index d768ab3..8a1df22 100644
--- a/src/3rdparty/chromium/build/common.gypi
+++ b/src/3rdparty/chromium/build/common.gypi
@@ -727,6 +727,9 @@
       # By default include non-appstore-compliant code.
       'appstore_compliant_code%': 0,
 
+      # Enable this to turn off the delete-null-pointer-checks optimization in GCC 6+
+      'no_delete_null_pointer_checks%': 0,
+
       'conditions': [
         # A flag for POSIX platforms
         ['OS=="win"', {
@@ -1271,6 +1274,7 @@
     'enable_hangout_services_extension%' : '<(enable_hangout_services_extension)',
     'proprietary_codecs%': '<(proprietary_codecs)',
     'appstore_compliant_code%': '<(appstore_compliant_code)',
+    'no_delete_null_pointer_checks%': '<(no_delete_null_pointer_checks)',
     'use_goma%': '<(use_goma)',
     'gomadir%': '<(gomadir)',
     'use_lto%': '<(use_lto)',
-- 
cgit v1.0-4-g1e03

From d6c8a2cf8fa374e6d1c0a578391b57112c047fa4 Mon Sep 17 00:00:00 2001
From: David Faure <david.faure@kdab.com>
Date: Mon, 5 Sep 2016 15:25:02 +0200
Subject: Enable -fno_delete_null_pointer_checks for g++ 6 on all of chromium

This is necessary e.g. for PaintLayer::enclosingSelfPaintingLayer
which also compares this with null.

Change-Id: I85d69432a0d7eeb0d8df8f395821880e36180dcc
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 src/core/config/linux.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/config/linux.pri b/src/core/config/linux.pri
index 9264932..a318e17 100644
--- a/src/core/config/linux.pri
+++ b/src/core/config/linux.pri
@@ -29,6 +29,8 @@ use?(nss) {
         use_openssl_certs=1
 }
 
+gcc:!clang: greaterThan(QT_GCC_MAJOR_VERSION, 5): GYP_CONFIG += no_delete_null_pointer_checks=1
+ 
 contains(QT_CONFIG, system-zlib): use?(system_minizip): GYP_CONFIG += use_system_zlib=1
 contains(QT_CONFIG, system-png): GYP_CONFIG += use_system_libpng=1
 contains(QT_CONFIG, system-jpeg): GYP_CONFIG += use_system_libjpeg=1
-- 
cgit v1.0-4-g1e03

